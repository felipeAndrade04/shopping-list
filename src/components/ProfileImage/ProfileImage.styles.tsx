import styled from 'styled-components/native';
import { ProfileImageProps } from './ProfileImage.types';

const sizeMap = {
  sm: 50,
  md: 50,
  lg: 148,
};

export const ProfileImageContainer = styled.View<Partial<ProfileImageProps>>`
  width: ${({ size }) => sizeMap[size]}px;
  height: ${({ size }) => sizeMap[size]}px;
  border-radius: 100%;
  background: ${({ theme }) => theme.colors.main};
`;

export const ProfileImage = styled.Image`
  flex: 1;
  border-radius: 100%;
`;

export const ProfileWithoutImage = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
`;

export const Text = styled.Text<Partial<ProfileImageProps>>`
  font-size: ${({ size }) => sizeMap[size] / 2}px;
  font-family: ${({ theme }) => theme.fonts.bold};
  color: ${({ theme }) => theme.colors.white};
`;
