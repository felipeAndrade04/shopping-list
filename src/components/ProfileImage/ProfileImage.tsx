import * as S from './ProfileImage.styles';
import { ProfileImageProps } from './ProfileImage.types';

export function ProfileImage({ size = 'md', url, name }: ProfileImageProps) {
  const firstLetter = name[0];

  return (
    <S.ProfileImageContainer size={size}>
      {url ? (
        <S.ProfileImage source={{ uri: url }} testID="profile-image" />
      ) : (
        <S.ProfileWithoutImage testID="profile-without-image">
          <S.Text size={size}>{firstLetter}</S.Text>
        </S.ProfileWithoutImage>
      )}
    </S.ProfileImageContainer>
  );
}
