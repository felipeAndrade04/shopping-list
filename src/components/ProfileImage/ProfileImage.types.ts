export interface ProfileImageProps {
  url?: string;
  size?: 'sm' | 'md' | 'lg';
  name: string;
}
