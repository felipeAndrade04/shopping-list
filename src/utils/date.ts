const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
