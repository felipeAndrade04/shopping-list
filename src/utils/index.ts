export * from './delay';
export * from './date';
