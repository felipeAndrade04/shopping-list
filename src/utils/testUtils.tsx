import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
