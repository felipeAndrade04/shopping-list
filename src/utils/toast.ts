import EventManager from './EvenManager';

export type ToastType = 'success' | 'info' | 'error';

export interface Toast {
  id?: string;
  type: ToastType;
  text: string;
  duration?: number;
}

export const toastEventManager = new EventManager();

export default function toast({ type, text, duration }: Toast) {
  toastEventManager.emit<Toast>('addtoast', {
    id: new Date().getTime().toString(),
    type,
    text,
    duration,
  });
}
