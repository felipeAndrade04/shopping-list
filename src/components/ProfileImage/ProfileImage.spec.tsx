import { render } from '@testing-library/react-native';
import { ProfileImage } from './ProfileImage';
import { wrapper } from '@app/utils/testUtils';

describe('<ProfileImage />', () => {
  const name = 'Felipe';
  const url = 'https://avatars.githubusercontent.com/u/42103489?v=4';

  it('Should be render correctly', () => {
    const tree = render(<ProfileImage name={name} />, { wrapper }).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should be render profile with image', () => {
    const { getByTestId } = render(<ProfileImage name={name} url={url} />, { wrapper });

    expect(getByTestId('profile-image')).toBeTruthy();
  });

  it('Should be render profile without image', () => {
    const { getByTestId, getByText } = render(<ProfileImage name={name} />, { wrapper });

    expect(getByText(name[0])).toBeTruthy();
    expect(getByTestId('profile-without-image')).toBeTruthy();
  });
});
