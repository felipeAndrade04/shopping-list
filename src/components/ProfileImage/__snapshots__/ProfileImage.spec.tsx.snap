// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileImage /> Should be render correctly 1`] = `
<View
  size="md"
  style={
    Array [
      Object {
        "backgroundColor": "#009Cf7",
        "borderRadius": "100%",
        "height": 50,
        "width": 50,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="profile-without-image"
  >
    <Text
      size="md"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Poppins-Bold",
            "fontSize": 25,
          },
        ]
      }
    >
      F
    </Text>
  </View>
</View>
`;
